ttnn/cpp/ttnn/operations/experimental/reshape/reshape_pybind.cpp

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "reshape_pybind.hpp"
+#include "reshape.hpp"
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "ttnn/cpp/pybind11/decorators.hpp"
+
+#include "ttnn/types.hpp"
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/tensor_impl.hpp"
+
+
+namespace ttnn::operations::experimental::reshape::detail {
+namespace py = pybind11;
+
+void py_bind_reshape(py::module& module) {
+    auto doc = R"doc(
+
+        Note: for a 0 cost view, the following conditions must be met:
+            * the last dimension must not change
+            * In Tiled the second last two dimensions must not change OR there is no padding on the second last dimension
```

Comment:
Can we add more to the description with respect to the unsafe nature?

Reply:
I agree, these are not the conditions for a 0 cost view, in the experimental version, these are the conditions for the function to not corrupt your data. The doc should say this and recommend the user uses ttnn.reshape if unsure which will behind the scenes call this function if it is safe to do so